server/app/services/auth-service.js

```diff
@@ -110,10 +110,59 @@ async function logout(tokenId) {
     return token;
 }
 
+/**
+ * Logout a member by revoking his token.
+ *
+ * @param tokenId Token's JIT
+ * @return {Promise.<JWT>} The deleted token
+ * @throws An error if the token could not be find.
+ */
+async function me(tokenId) {
+    const token = await JWT.findById(tokenId);
+
+    const barman = await Barman.findOne({
+        include: [{
+            model: ConnectionInformation,
+            as: 'connection',
+            include: [{
+                model: JWT,
+                as: 'jwt',
+                where: {id: token.id},
+                attributes: []
+            }],
+            attributes: ['username']
+        }, {
+            model: Kommission,
+            as: 'kommissions'
+        }, {
+            model: Role,
+            as: 'roles'
+        }]
+    });
+
+    const specialAccount = await SpecialAccount.findOne({
+        include: [{
+            model: ConnectionInformation,
+            as: 'connection',
+            include: [{
+                model: JWT,
+                as: 'jwt',
+                where: {id: token.id},
+                attributes: []
+            }],
+            attributes: ['username'],
+        }],
+        attributes: ['id', 'createdAt', 'description', 'updatedAt']
+    });
```

Comment:
A token is already linked to a Barman or a SpecialAccount,
Just use `token.getBarman()` and/or `token.getSpecialAccount()` to get the related user.
Last thing is to correctly remove unwanted fields

Reply:
I try with `token.getBarman()` and it send me back an internal server error. So I think we have to use my way.
What do you mean about correctly remove unwanted fields ?